src/client/minimap.cpp

```diff
@@ -461,6 +444,29 @@ void Minimap::blitMinimapPixelsToImageSurface(
 	}
 }
 
+video::IImage *Minimap::getMinimapMask()
```

Comment:
> Couldn't you simplify this by having just one `data->minimap_mask` field (which holds either the round or the square texture)?

Only if you recreate the mask every time `data->minimap_shape_round` changes, so without causing performance impacts you can't.

> It also feels slightly odd to me having this texture initialization be detached from "the other" texture initialisation guarded by `data->textures_initialised`. What's the reason for that?

It may be the case that only `data->minimap_mask_round` or only `data->minimap_mask_square` is initialized.
Also, they aren't textures, but images created from a texture.

`date->texture` does not get initialized since it is not static and changes dynamically.

I'm not the original author of the minimap code, so I don't know how necessary it is (performance-wise) that any of the textures get cached, but it probably has its reason and I didn't want to change it.

Reply:
> Only if you recreate the mask every time `data->minimap_shape_round` changes, so without causing performance impacts you can't.

I think it would be safe to assume that this shouldn't change at runtime since it is a setting (though then we should cache its value to be sure it actually doesn't).

Anyways, thanks for the answers, this is fine by me.

Reply:
> I think it would be safe to assume that this shouldn't change at runtime since it is a setting

`minimap_shape_round` changes when you press <kbd>Shift</kbd>+<kbd>V</kbd>, I think

Reply:
> `minimap_shape_round` changes when you press Shift+V, I think

yes it does, from circular to rectangle and back

Reply:
lol. thanks for the info. is this documented anywhere?

Reply:
yes, it's documented in README.md, for example

grep for "Change minimap orientation"